src/nspawn/nspawn.c

```diff
@@ -3848,6 +4053,19 @@ int main(int argc, char *argv[]) {
                         goto finish;
                 }
 
+                r = sd_event_new(&event);
+                if (r < 0) {
+                        log_error_errno(r, "Failed to get default event source: %m");
+                        goto finish;
+                }
+
+                if (arg_notify_mode != NOTIFY_NO) {
+                        r = setup_sd_notify_parent(event, notify_socket, &pid);
```

Comment:
your function parameter is an "int*", but pid is a pid_t. Please use the right type everywhere.